File: models/bloodbank/BloodRequest.js
// models/BloodRequest.js
import mongoose from "mongoose";

const BloodRequestSchema = new mongoose.Schema(
  {
    hospitalId: { type: mongoose.Schema.Types.ObjectId, ref: "Hospital", required: true },
    branchId: { type: mongoose.Schema.Types.ObjectId, ref: "Branch", required: true },

    patientId: { type: mongoose.Schema.Types.ObjectId, ref: "Patient", required: true },

    requestedBy: { type: mongoose.Schema.Types.ObjectId, ref: "User" },

    bloodGroup: { type: mongoose.Schema.Types.ObjectId, ref: "BloodGroup" },
    componentType: String,
    quantityUnits: Number,

    status: {
      type: String,
      enum: ["Pending", "Approved", "Issued", "Rejected"],
      default: "Pending",
    },
  },
  { timestamps: true }
);

export default mongoose.model("BloodRequest", BloodRequestSchema);

File: jstojv/src/main/java/com/samrat/jstojv/entity/BloodRequest.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "blood_requests")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class BloodRequest extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hospital_id", nullable = false)
    private Hospital hospital;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "branch_id", nullable = false)
    private Branch branch;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "patient_id", nullable = false)
    private PatientRegistration patient;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "requested_by")
    private User requestedBy;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "blood_group_id")
    private BloodGroup bloodGroup;

    @Column(name = "component_type")
    private String componentType;

    @Column(name = "quantity_units")
    private Integer quantityUnits;

    @Enumerated(EnumType.STRING)
    private Status status = Status.Pending;

    public enum Status {
        Pending, Approved, Issued, Rejected
    }
}
